sensors/EncoderSensor.py

```diff
@@ -0,0 +1,58 @@
+from src.services.Encoder import Encoder
+from src.pin.pinout import Pinout
+from src.pin import memory
+from src.shared_memory import SharedMemory
+from test_lib.input.aggregate import Multiple
+from test_lib.input.input import Input, InputFailedException
+from test_lib.condition.condition import Condition
+from test_lib.condition.checked import Checked
+
+COUNTER_DISTANCE_IN_METERS = 0.0001
+N_FRAMES = 100
+FRAME_SIZE_IN_SECONDS = 0.005
+START_COUNTER = 2**31 - 1
+
+class EncoderSensor():
+    def __init__(self,shm:SharedMemory,pin1:Pinout,pin2:Pinout):
+        self._encoder=Encoder(shm,pin1,pin2)
+        self.position=0
+        self.direction=memory.Encoder.Direction.Forwards
+        self.speed=0.0
+        self.acceleration=0.0
+        self.counter=0
+
+    class EncoderSensorException(InputFailedException):
+        def __str__(self):
+            return "The encoder is not on"
+    
+    def _check_pin_state(self) -> bool:
+        if not self._encoder.get_is_on():
+            raise self.EncoderSensorException()
+        else:
+            return True
+
+    def generate_speed(self,direction:memory.Encoder.Direction.value,speed:float,acceleration:float)->Input:
+        if not self._encoder.get_is_on():
+            raise self.EncoderSensorException()
```

Comment:
Same as with linear sensor, this check should be done by the `Checked` input, because the user might prepare the input and apply it later during execution, and in that time the state might have changed

Reply:
notice, this is also in the generate_counter function